compiler/qsc_partial_eval/src/lib.rs

```diff
@@ -525,6 +525,159 @@ impl<'a> PartialEvaluator<'a> {
+                    ty: rir::Ty::Boolean,
+                };
+                let rhs_eval_condition_ins = Instruction::Icmp(
+                    ConditionCode::Eq,
+                    Operand::Variable(lhs_rir_var),
+                    Operand::Literal(Literal::Bool(false)),
+                    rhs_eval_condition_rir_var,
+                );
+                self.get_current_rir_block_mut()
+                    .0
+                    .push(rhs_eval_condition_ins);
+                self.eval_logical_bool_bin_op(true, rhs_eval_condition_rir_var, rhs_expr_id)?
+            }
+            _ => panic!("invalid Boolean operator {bin_op:?}"),
+        };
+        Ok(EvalControlFlow::Continue(Value::Var(result_var)))
+    }
+
+    fn eval_comparison_bool_bin_op(
+        &mut self,
+        bin_op: BinOp,
+        lhs_eval_var: Var,
+        rhs_expr_id: ExprId,
+    ) -> Result<Var, Error> {
+        // Try to evaluate the RHS expression to get its value and create a RHS operand.
+        let rhs_control_flow = self.try_eval_expr(rhs_expr_id)?;
+        let EvalControlFlow::Continue(rhs_value) = rhs_control_flow else {
+            let rhs_expr = self.get_expr(rhs_expr_id);
+            return Err(Error::Unexpected(
+                "embedded return in RHS expression".to_string(),
+                rhs_expr.span,
+            ));
+        };
+        let rhs_operand = map_eval_value_to_rir_operand(&rhs_value);
+
+        // Get the comparison result depending on the operator and the RHS value.
+        let result_var = match (bin_op, rhs_operand) {
+            // If the RHS value is a literal, depending on the operand, the result of the Boolean comparison is just the
+            // LHS value.
+            (BinOp::Neq, Operand::Literal(Literal::Bool(false)))
+            | (BinOp::Eq, Operand::Literal(Literal::Bool(true))) => lhs_eval_var,
+            // In other cases we have to actually generate the comparison instruction.
+            (BinOp::Eq | BinOp::Neq, _) => {
+                let rir_variable = rir::Variable::new_boolean(self.resource_manager.next_var());
+                let lhs_operand = Operand::Variable(map_eval_var_to_rir_var(lhs_eval_var));
+                let condition_code = match bin_op {
+                    BinOp::Eq => ConditionCode::Eq,
+                    BinOp::Neq => ConditionCode::Ne,
+                    _ => panic!("invalid Boolean comparison operator {bin_op:?}"),
+                };
+                let cmp_inst =
+                    Instruction::Icmp(condition_code, lhs_operand, rhs_operand, rir_variable);
+                self.get_current_rir_block_mut().0.push(cmp_inst);
+                map_rir_var_to_eval_var(rir_variable)
+            }
+            (_, _) => panic!("invalid Boolean comparison operator {bin_op:?}"),
+        };
+        Ok(result_var)
+    }
+
+    fn eval_logical_bool_bin_op(
+        &mut self,
+        default_result: bool,
+        rhs_eval_condition_var: rir::Variable,
```

Comment:
Consider changing the name to something that conveys that this comes from the LHS of the expression.

Reply:
Not needed anymore.